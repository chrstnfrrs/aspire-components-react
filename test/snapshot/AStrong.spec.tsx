import { AStrong } from '../../src/AStrong/AStrong';
import { AS_BODY_ENUM } from '../enums/as-body-enum';
import { FONTSIZE_ENUM } from '../enums/fontsize-enum';
import { TYPE_BODY_ENUM } from '../enums/type-body-enum';

import { testComponent } from './utils/test-component';
import { testTypography } from './utils/test-typography';

describe('Given AStrong component is used', () => {
  const AStrongProps = {
    factory: {
      fontSize: FONTSIZE_ENUM,
    },
  };

  testComponent(AStrong, 'AStrong', AStrongProps);
  testTypography(AStrong, 'AStrong', 'as', AS_BODY_ENUM);
  testTypography(AStrong, 'AStrong', 'type', TYPE_BODY_ENUM);
});
