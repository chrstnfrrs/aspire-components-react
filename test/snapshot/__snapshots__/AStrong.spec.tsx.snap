// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given AStrong component is used When a developer uses AStrong with a className Then the the class should be present 1`] = `
<strong
  className="asdf"
>
  AStrong
</strong>
`;

exports[`Given AStrong component is used When a developer uses AStrong with no props Then the only class should be a-container 1`] = `
<strong
  className=""
>
  AStrong
</strong>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 2xl Then the class fontSize-2xl should be used 1`] = `
<strong
  className="fontSize-2xl"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 3xl Then the class fontSize-3xl should be used 1`] = `
<strong
  className="fontSize-3xl"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 4xl Then the class fontSize-4xl should be used 1`] = `
<strong
  className="fontSize-4xl"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 5xl Then the class fontSize-5xl should be used 1`] = `
<strong
  className="fontSize-5xl"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 6xl Then the class fontSize-6xl should be used 1`] = `
<strong
  className="fontSize-6xl"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 7xl Then the class fontSize-7xl should be used 1`] = `
<strong
  className="fontSize-7xl"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 8xl Then the class fontSize-8xl should be used 1`] = `
<strong
  className="fontSize-8xl"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 9xl Then the class fontSize-9xl should be used 1`] = `
<strong
  className="fontSize-9xl"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in an invalid value Then no fontSize class should be used 1`] = `
<strong
  className=""
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in base Then the class fontSize-base should be used 1`] = `
<strong
  className="fontSize-base"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in lg Then the class fontSize-lg should be used 1`] = `
<strong
  className="fontSize-lg"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in sm Then the class fontSize-sm should be used 1`] = `
<strong
  className="fontSize-sm"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in undefined Then no fontSize class should be used 1`] = `
<strong
  className=""
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in xl Then the class fontSize-xl should be used 1`] = `
<strong
  className="fontSize-xl"
/>
`;

exports[`Given AStrong component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in xs Then the class fontSize-xs should be used 1`] = `
<strong
  className="fontSize-xs"
/>
`;

exports[`Given AStrong component is used When rendering AStrong with as set to "body-1" Then the component should render as expected 1`] = `
<strong
  className="a-body-1"
>
  AStrong
</strong>
`;

exports[`Given AStrong component is used When rendering AStrong with as set to "body-2" Then the component should render as expected 1`] = `
<strong
  className="a-body-2"
>
  AStrong
</strong>
`;

exports[`Given AStrong component is used When rendering AStrong with as set to "button-text" Then the component should render as expected 1`] = `
<strong
  className="a-button-text"
>
  AStrong
</strong>
`;

exports[`Given AStrong component is used When rendering AStrong with as set to "caption" Then the component should render as expected 1`] = `
<strong
  className="a-caption"
>
  AStrong
</strong>
`;

exports[`Given AStrong component is used When rendering AStrong with as set to "overline" Then the component should render as expected 1`] = `
<strong
  className="a-overline"
>
  AStrong
</strong>
`;

exports[`Given AStrong component is used When rendering AStrong with type set to "em" Then the component should render as expected 1`] = `
<em
  className=""
>
  AStrong
</em>
`;

exports[`Given AStrong component is used When rendering AStrong with type set to "p" Then the component should render as expected 1`] = `
<p
  className=""
>
  AStrong
</p>
`;

exports[`Given AStrong component is used When rendering AStrong with type set to "span" Then the component should render as expected 1`] = `
<span
  className=""
>
  AStrong
</span>
`;

exports[`Given AStrong component is used When rendering AStrong with type set to "strong" Then the component should render as expected 1`] = `
<strong
  className=""
>
  AStrong
</strong>
`;
